homeassistant/components/sia/hub.py

```diff
@@ -0,0 +1,90 @@
+"""The sia hub."""
+import logging
+
+from pysiaalarm.aio import CommunicationsProtocol, SIAAccount, SIAClient, SIAEvent
+
+from homeassistant.const import CONF_PORT, CONF_PROTOCOL, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, EventOrigin, HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_IGNORE_TIMESTAMPS,
+    DOMAIN,
+    SIA_EVENT,
+)
+
+DEFAULT_TIMEBAND = (80, 40)
+IGNORED_TIMEBAND = (3600, 1800)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SIAHub:
+    """Class for SIA Hubs."""
+
+    def __init__(
+        self, hass: HomeAssistant, hub_config: dict, entry_id: str, title: str
+    ):
+        """Create the SIAHub."""
+        self._hass = hass
+        self._port = int(hub_config[CONF_PORT])
+        self.entry_id = entry_id
+        self._title = title
+        self._accounts = hub_config[CONF_ACCOUNTS]
+
+        self._remove_shutdown_listener = None
+        self.sia_accounts = [
+            SIAAccount(
+                account_id=a[CONF_ACCOUNT],
+                key=a.get(CONF_ENCRYPTION_KEY),
+                allowed_timeband=IGNORED_TIMEBAND
+                if a[CONF_IGNORE_TIMESTAMPS]
+                else DEFAULT_TIMEBAND,
+            )
+            for a in self._accounts
+        ]
+        self.sia_client = SIAClient(
+            host="",
+            port=self._port,
+            accounts=self.sia_accounts,
+            function=self.async_create_and_fire_event,
+            protocol=CommunicationsProtocol(hub_config[CONF_PROTOCOL]),
+        )
```

Comment:
Please remove side effects from `__init__`

Set these up in `async_setup_hub` instead of creating them in `__init__`

Reply:
Done, but both of those are also questionable of what a side effect is, they both do nothing by initializing a variable, a bit more complex then a direct assignment but still.